File: testdata/fixtures/ai/python/detect-mistral.py
# AI usage detection: Mistral
def vuln():
    # ruleid: detect-mistral
    import mistralai
    # ruleid: detect-mistral
    client = MistralClient(api_key="key")
    # ruleid: detect-mistral
    resp = client.chat(model="mistral-large", messages=[])


def safe():
    # ok: detect-mistral
    import os
    # ok: detect-mistral
    client = OtherClient()
    # ok: detect-mistral
    data = "safe string"

File: testdata/fixtures/ai/typescript/detect-mistral.ts
// AI usage detection: Mistral
// ruleid: detect-mistral
import "@mistralai";
// ruleid: detect-mistral
import MistralClient from "@mistralai";
// ruleid: detect-mistral
const mc = new MistralClient("key");
// ruleid: detect-mistral
const r = client.chat({ model: "mistral-large" });
// ok: detect-mistral
import fs from "fs";
// ok: detect-mistral
const text = "plain string";
// ok: detect-mistral
console.log("hello");
